keel-web/src/main/kotlin/com/netflix/spinnaker/keel/rest/DeliveryConfigController.kt

```diff
@@ -75,19 +79,23 @@ class DeliveryConfigController(
     produces = [APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE]
   )
   @PreAuthorize("@authorizationSupport.hasApplicationPermission('READ', 'APPLICATION', #deliveryConfig.application)")
-  fun diff(@RequestBody deliveryConfig: SubmittedDeliveryConfig): List<EnvironmentDiff> =
-    adHocDiffer.calculate(deliveryConfig)
+  fun diff(@RequestBody deliveryConfig: SubmittedDeliveryConfig): List<EnvironmentDiff> {
+    validator.validate(deliveryConfig)
+    return adHocDiffer.calculate(deliveryConfig)
+  }
 
   @PostMapping(
     path = ["/validate"],
     consumes = [APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE],
     produces = [APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE]
   )
   @PreAuthorize("@authorizationSupport.hasApplicationPermission('READ', 'APPLICATION', #deliveryConfig.application)")
-  fun validate(@RequestBody deliveryConfig: SubmittedDeliveryConfig) =
-  // TODO: replace with JSON schema/OpenAPI spec validation when ready (for now, leveraging parsing error handling
+  fun validate(@RequestBody deliveryConfig: SubmittedDeliveryConfig) {
+    // TODO: replace with JSON schema/OpenAPI spec validation when ready (for now, leveraging parsing error handling
     //  in [ExceptionHandler])
-    mapOf("status" to "valid")
+    validator.validate(deliveryConfig)
+    // mapOf("status" to "valid")
```

Comment:
I think we can remove `mapOf("status" to "valid")` here and just return 200 if its valid?

Reply:
I think [204](https://httpstatuses.com/204) would be a more appropriate status code if we aren't returning a body, but, otherwise, I agree we don't need to return anything in the body.

Reply:
Yeah, the trick is matching that to the return type of the function... Maybe if you make it nullable it just works?

Reply:
Never mind, I see you changed the return type of the function. Sure, go ahead and remove the map and change the status code to 204. 👍